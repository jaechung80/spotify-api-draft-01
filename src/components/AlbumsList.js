import React, { useEffect } from 'react';
import { connect } from 'react-redux';

const AlbumsList = (props) => {
  useEffect(() => {
    console.log('in AlbumsList props.albums', props.albums);
  });

  return (
    <div>
      <ul>
        {props.albums?.items?.map((album) => {
          return (
            <div>
              <li>Artist: {album.artists[0].name}</li>
              <li>Album: {album.name}</li>
              <li>
                <img src={album.images[0].url} />
              </li>
            </div>
          );
        })}
      </ul>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    albums: state.albums,
  };
};

export default connect(mapStateToProps)(AlbumsList);
