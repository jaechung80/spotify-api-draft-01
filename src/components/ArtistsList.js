import React, { useEffect } from 'react';
import { Card, Container, Row, Col } from 'react-bootstrap';

const ArtistsList = (props) => {
  useEffect(() => {
    console.log('in ArtistsList props.artists', props.artists);
  });

  return (
    <Container>
      <Row>
        {props.artists?.items?.map((artist) => {
          return (
            <Col xs='3'>
              <div className='artists'>
                <Card key={artist.id} style={{ width: '18rem' }}>
                  <Card.Img
                    variant='top'
                    src={
                      artist.images.length > 0
                        ? artist.images[0].url
                        : 'https://upload.wikimedia.org/wikipedia/commons/2/26/512pxIcon-sunset_photo_not_found.png'
                    }
                  />
                  <Card.Body>
                    <Card.Title>{artist.name}</Card.Title>
                    <Card.Text>
                      Genres: {artist.genres[0] || 'not listed'}
                    </Card.Text>
                  </Card.Body>
                </Card>
              </div>
            </Col>
          );
        })}
      </Row>
    </Container>
  );
};

export default ArtistsList;
