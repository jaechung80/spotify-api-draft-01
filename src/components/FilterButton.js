import React from 'react';
import Button from 'react-bootstrap/Button';

const FilterButton = (props) => {
  const { onClick, filter } = props;
  return <Button onClick={onClick}>{filter}</Button>;
};

export default FilterButton;
