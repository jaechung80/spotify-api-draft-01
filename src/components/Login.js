import React from 'react';

const Home = () => {
  const {
    REACT_APP_CLIENT_ID,
    REACT_APP_AUTHORIZE_URL,
    REACT_APP_REDIRECT_URL,
  } = process.env;

  const handleLogin = () => {
    window.location = `${REACT_APP_AUTHORIZE_URL}?client_id=${REACT_APP_CLIENT_ID}&redirect_uri=${REACT_APP_REDIRECT_URL}&response_type=token&show_dialog=true`;
  };

  return (
    <div className='login'>
      <button type='submit' onClick={handleLogin}>
        Login to spotify
      </button>
    </div>
  );
};

export default Home;
