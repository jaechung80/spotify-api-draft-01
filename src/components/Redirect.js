import { useEffect } from 'react';
import _ from 'lodash';
import { getParamValues } from '../api/index';

const RedirectPage = (props) => {
  useEffect(() => {
    const { history, location } = props;
    try {
      if (_.isEmpty(location.hash)) {
        return history.push('/home');
      }
      const access_token = getParamValues(location.hash);
      const expiryTime = new Date().getTime() + access_token.expires_in * 1000;
      localStorage.setItem('params', JSON.stringify(access_token));
      localStorage.setItem('expiry_time', expiryTime);
      console.log('in redirect, localStorage successful', localStorage);
      history.push('/home');
    } catch (error) {
      console.log('error in redirect catch', error);
      history.push('/');
    }
  });

  return null;
};

export default RedirectPage;
