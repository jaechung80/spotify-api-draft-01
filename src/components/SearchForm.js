import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { searchSpotify } from '../store/albums';

function SearchForm(props) {
  const [search, setSearch] = useState('');

  const handleChange = (evt) => {
    setSearch(evt.target.value);
  };

  const handleSearch = (evt) => {
    evt.preventDefault();
    console.log('in handleSearch before anything happens');
    props.searchSpotify(search);
  };

  useEffect(() => console.log('here is search in state', search));

  return (
    <div>
      <form onSubmit={handleSearch}>
        <input
          type='text'
          name='search'
          value={search}
          onChange={handleChange}
        />
        <button type='submit'>Search</button>
      </form>
    </div>
  );
}

const mapDispatchToProps = (dispatch) => {
  return {
    searchSpotify: (search) => dispatch(searchSpotify(search)),
  };
};

export default connect(null, mapDispatchToProps)(SearchForm);
