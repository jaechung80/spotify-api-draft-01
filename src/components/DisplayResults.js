import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import AlbumsList from './AlbumsList';

const DisplayResults = (props) => {
  useEffect(() => {
    console.log('in DisplayResults props.albums', props.albums);
  });

  return (
    <div>
      test
      <AlbumsList />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    albums: state.albums,
  };
};

export default connect(mapStateToProps)(DisplayResults);
