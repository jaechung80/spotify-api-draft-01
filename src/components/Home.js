import React from 'react';
import { SearchForm } from './SearchForm';
import { connect } from 'react-redux';
import searchSpotify from '../store/common';

const Home = (props) => {
  const handleSearch = (search) => {
    console.log('in handleSearch before anything happens');
    props.searchSpotify(search);
  };

  return (
    <>
      <div>Home</div>
      <SearchForm handleSearch={handleSearch} />
    </>
  );
};

export const mapDispatchToProps = (dispatch) => {
  return {
    searchSpotify: (search) => dispatch(searchSpotify(search)),
  };
};

export default connect(null, mapDispatchToProps)(Home);
