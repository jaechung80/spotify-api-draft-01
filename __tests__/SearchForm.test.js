import React from 'react';
import { SearchForm } from '../src/components/SearchForm';
import { shallow, mount } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('SearchForm component', () => {
  it('renders SearchForm component correctly', () => {
    const wrapper = shallow(<SearchForm />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SearchForm search field', () => {
  const wrapper = shallow(<SearchForm />);

  it('renders the text input box', () => {
    expect(wrapper.find('input').exists()).toBe(true);
  });

  it('starts out as an empty string', () => {
    expect(wrapper.find('input').prop('value')).toEqual('');
  });

  it('should have proper props for text input field', () => {
    expect(wrapper.find('input').props()).toEqual({
      id: 'search',
      name: 'search',
      onChange: expect.any(Function),
      type: 'text',
      value: '',
    });
  });

  it('should set the search field value on change event', () => {
    wrapper.find('input').simulate('change', {
      target: {
        value: 'amy',
      },
    });
    expect(wrapper.find('input').prop('value')).toEqual('amy');
  });
});

describe('SearchForm submit button', () => {
  let wrapper;
  let mockSubmit;

  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(<SearchForm handleSearch={mockSubmit} />);
  });

  it('renders the submit button correctly', () => {
    expect(wrapper.find('button').exists()).toBe(true);
  });

  it('submits the search when clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockSubmit).toHaveBeenCalled();
  });
});
