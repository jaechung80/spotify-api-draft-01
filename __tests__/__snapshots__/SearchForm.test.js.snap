// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders SearchForm component correctly 1`] = `ShallowWrapper {}`;

exports[`SearchForm component renders SearchForm component without crashing 1`] = `ShallowWrapper {}`;
